src/raftstore/store/read_queue.rs

```diff
@@ -0,0 +1,266 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::collections::VecDeque;
+use std::{cmp, usize};
+
+use crate::raftstore::store::fsm::apply;
+use crate::raftstore::store::metrics::*;
+use crate::raftstore::store::Callback;
+
+use engine_rocks::RocksEngine;
+use kvproto::raft_cmdpb::RaftCmdRequest;
+use tikv_util::collections::HashMap;
+use tikv_util::time::{duration_to_sec, monotonic_raw_now};
+use tikv_util::MustConsumeVec;
+use time::Timespec;
+use uuid::Uuid;
+
+const READ_QUEUE_SHRINK_SIZE: usize = 64;
+
+pub(super) struct ReadIndexRequest {
```

Comment:
Why mark it as `super`?

Reply:
The queue shouldn't be accessable by other mods.